satpy/modifiers/angles.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilties for getting various angles for a dataset.."""
+from __future__ import annotations
+
+import os
+import hashlib
+from datetime import datetime
+from functools import update_wrapper
+from glob import glob
+from typing import Callable, Any, Optional, Union
+import shutil
+
+import dask
+import dask.array as da
+import numpy as np
+import xarray as xr
+from satpy.utils import get_satpos, ignore_invalid_float_warnings
+from pyorbital.orbital import get_observer_look
+from pyorbital.astronomy import get_alt_az, sun_zenith_angle
+from pyresample.geometry import SwathDefinition, AreaDefinition
+
+import satpy
+
+PRGeometry = Union[SwathDefinition, AreaDefinition]
+
+# Arbitrary time used when computing sensor angles that is passed to
+# pyorbital's get_observer_look function.
+# The difference is on the order of 1e-10 at most as time changes so we force
+# it to a single time for easier caching. It is *only* used if caching.
+STATIC_EARTH_INERTIAL_DATETIME = datetime(2020, 1, 1, 0, 0, 0)
+# simple integer for future proofing function changes that would invalidate
+# existing files in a user's cache
+_CACHE_VERSION = 1
+
+
+class ZarrCacheHelper:
+    """Helper for caching function results to on-disk zarr arrays."""
+
+    def __init__(self,
+                 func: Callable,
+                 cache_config_key: str,
+                 uncacheable_arg_types=(SwathDefinition, xr.DataArray, da.Array),
+                 sanitize_args_func: Callable = None,
+                 ):
+        """Hold on to provided arguments for future use."""
+        self._func = func
+        self._cache_config_key = cache_config_key
+        self._uncacheable_arg_types = uncacheable_arg_types
+        self._sanitize_args_func = sanitize_args_func
+
+    def cache_clear(self, cache_dir: Optional[str] = None):
```

Comment:
`cache_clear` or `clear_cache`? The latter reads better to me.

Reply:
`cache_clear` matches what the `lru_cache` in functools uses: https://docs.python.org/3/library/functools.html#functools.lru_cache